Compile error with heap and mutable string

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.04
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1) 10.3.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-01-07 04:37:44
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 c1711b8.60527a5

Git version: git version 2.30.2
Git vroot status: weekly.2021.52.1-74-g60527a56
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
Similar too #12998

`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/structs/ddd.v`
```v
module main

[heap]
pub struct Grid {
pub mut:
	header      []string
}

fn  main() {
    h := ['yore', 'yaya']

    mut grid := Grid{			
        header: h
    }
    wrap_text(mut grid)
}

fn wrap_text(mut gv Grid) {

    for mut ch in gv.header {
      ch = ch[1..2]
    }

    println(gv)
}
```

**What did you expect to see?**

No compile error

```
Grid{
    header: ['o', 'a']
}
```

**What did you see instead?**
```
==================
/tmp/v_1000/ddd.8789599844591507568.tmp.c:10966: error: cannot convert 'struct string *' to 'struct string'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -65,6 +65,9 @@ fn (mut g Gen) range_expr(node ast.IndexExpr, range ast.RangeExpr) {
 		} else {
 			g.write('string_substr(')
 		}
+		if node.left_type.is_ptr() {
+			g.write('*')
+		}
 		g.expr(node.left)
 	} else if sym.kind == .array {
 		if node.is_gated {

--- a/vlib/v/tests/string_index_in_for_mut_in_test.v
+++ b/vlib/v/tests/string_index_in_for_mut_in_test.v
@@ -1,0 +1,25 @@
+module main
+
+[heap]
+pub struct Grid {
+pub mut:
+	header []string
+}
+
+fn test_string_index_in_for_mut_in() {
+	h := ['yore', 'yaya']
+
+	mut grid := Grid{
+		header: h
+	}
+	wrap_text(mut grid)
+}
+
+fn wrap_text(mut gv Grid) {
+	for mut ch in gv.header {
+		ch = ch[1..2]
+	}
+
+	println(gv)
+	assert gv.header == ['o', 'a']
+}

